### DISCUSSION TITLE
Setting extended attribute doesn't work - and returns no error

### DISCUSSION THREAD

**Discussion by pjotrek-b** [waltligon/orangefs#105] [Q&A] [answered]
Hello everyone :smiley:

This is my very first installation with OrangeFS.
I've followed [the OrangeFS documentation for package installation](https://docs.orangefs.com/quickstart/quickstart-package/) (on Fedora 39) - for a single node simple test-setup.
I can successfully mount Orangefs (`/mnt/ofs`) and upload files.

Great!

However, trying to set/get extended metadata attributes as mentioned in [In the FAQs: '5.12 What are extended attributes? How do I use them with orangefs?'](http://www.orangefs.org/faq/#item5.12).

`$ setfattr -n key1 -v val1 /mnt/ofs/cover.jpg`

Returns no error message and exists with 0 (=good).

Reading the key fails though, claiming it has never been set:

`$ getfattr -n key1 /mnt/ofs/cover.jpg`

> /mnt/ofs/cover.jpg: key1: No such attribute

These are the default packages on Fedora39, installed and setup all according to the above mentioned documentation.
Any help/input is greatly appreciated.

Thank you very much in advance.

**Comment by hubcapsc** [comment, marked as answer]
I remember something about namespace...

This doesn't work for me on Fedora 39:
setfattr -n foo3 -v foo3val /pvfsmnt/file1

This does work for me on Fedora 39:
setfattr -n user.foo3 -v foo3val /pvfsmnt/file1

-Mike

**Reply by pjotrek-b** [reply]
Oh! Namespaces.
That was it! Thank you very much!
Namespaces and prefix are mentioned in the FAQ, but I assumed the examples were copy/paste safe?

I now tested storing the MD5 hashcode as extended attribute:
`$ setfattr -n user.checksum -v "$(md5sum cover.jpg)" cover.jpg`

works! :fireworks: 

Maybe the FAQs could be updated to contain an example that works as-is?

For example, this works:
`$ setfattr -n user.key1 -v "value 1" cover.jpg`

**Comment by waltligon** [comment]
Yes, there are a few namespaces but the one they want is “user”

**Comment by waltligon** [comment]
And the documentation should say this

**Reply by pjotrek-b** [reply]
Sorry if I overlooked it...?

**Comment by waltligon** [comment]
No, I said it should, but it might not. We need to fix that
